### ISSUE TITLE
[Bug] partialBundling does not work properly when output to esm format

### ISSUE THREAD

**Issue by niuiic** [farm-fe/farm#741]
### Steps to reproduce

```sh
git clone https://github.com/niuiic/fx-flow -b feat/pack-in-chunks
pnpm initialize
pnpm build
```

Copy `packages/core/dist` to another project and write a test file.

```
mkdir test
cd test
pnpm init
pnpm add fx-flow
# replace node_modules/fx-flow/dist with the built one
```

`index.mjs`
```typescript
import { add } from "fx-flow";

console.log(add(1, 2));
```

Related to #709, `output.targetEnv = "node"` works for cjs format, but not esm.

### What is expected?

Functions can be imported properly.

### What is actually happening?

```
file:///home/niuiic/Downloads/test/index.mjs:3
console.log(add(1, 2));
            ^

TypeError: add is not a function
    at file:///home/niuiic/Downloads/test/index.mjs:3:13
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)

Node.js v21.1.0
```

Set `output.targetEnv` to `node`

```
node:internal/modules/cjs/loader:1408
      const err = new ERR_REQUIRE_ESM(filename, usesEsm, parentPath,
                  ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/flow_andThen.75ed8fd1.js from /home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/index.mjs not supported.
flow_andThen.75ed8fd1.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead either rename flow_andThen.75ed8fd1.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/index.mjs:1:10811 {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v21.1.0
```

### System Info

```Shell
System:
    OS: Linux 6.1 Gentoo Linux
    CPU: (16) x64 AMD Ryzen 7 5800H with Radeon Graphics
    Memory: 10.53 GB / 15.59 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.1.0 - /run/user/1000/fnm_multishells/191360_1700186487218/bin/node
    npm: 10.2.0 - /run/user/1000/fnm_multishells/191360_1700186487218/bin/npm
    pnpm: 8.10.5 - /run/user/1000/fnm_multishells/191360_1700186487218/bin/pnpm
```
```


### Any additional comments?

BTW, do you plan to support dynamic import in core, or implement it with a plugin?

**Comment by wre232114** [comment]
It will be fixed in 0.14 today.

**Comment by niuiic** [comment]
I tried v0.14. New issues were found.

1. Just follow the previous steps.

```
ReferenceError: location is not defined
    at Object.moduleCreated (file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/__farm_runtime.mjs:1:8023)
    at i.hookSerial (file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/__farm_runtime.mjs:1:3817)
    at d.require (file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/__farm_runtime.mjs:1:1788)
    at file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/index.mjs:1:5350
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)
```

2. Run `pnpm build --filter fx-flow --force` two times.

```
> @ build /home/niuiic/Documents/projects/node/fx-flow
> turbo build "--filter" "fx-flow" "--force"

• Packages in scope: fx-flow
• Running build in 1 packages
• Remote caching disabled
fx-flow:build: cache bypass, force executing 944fccf30bc0be06
fx-flow:build:
fx-flow:build: > fx-flow@0.4.2 build /home/niuiic/Documents/projects/node/fx-flow/packages/core
fx-flow:build: > farm build && farm build -c farm.config.cjs.ts
fx-flow:build:
fx-flow:build: [ Farm ] Using config file at /home/niuiic/Documents/projects/node/fx-flow/packages/core/farm.config.ts
fx-flow:build: thread '<unnamed>' panicked at crates/plugin_script/src/handle_entry_resources.rs:409:8:
fx-flow:build: runtime resource is not found
fx-flow:build: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fx-flow:build:  ELIFECYCLE  Command failed with exit code 134.
fx-flow:build: ERROR: command finished with error: command (/home/niuiic/Documents/projects/node/fx-flow/packages/core) pnpm run build exited (1)
fx-flow#build: command (/home/niuiic/Documents/projects/node/fx-flow/packages/core) pnpm run build exited (1)

 Tasks:    0 successful, 1 total
Cached:    0 cached, 1 total
  Time:    3.781s
Failed:    fx-flow#build

 ERROR  run failed: command  exited (1)
 ELIFECYCLE  Command failed with exit code 1.
```
